pyproject.toml

```diff
@@ -21,3 +21,77 @@ commands = pytest
 extras = dev
 """
 
+[tool.ruff]
+# Enable Pyflakes `E` and `F` codes by default.
+select = [
+    "F",  # Pyflakes
+    "E",  # pycodestyle (error)
+    "W",    # pycodestyle (warning)
+    # "C90",  # mccabe
+    "I",    # isort
+    "D",    # pydocstyle
+    # "PL",   # Pylint
```

Comment:
I would enable these as well. Pylint is complainy, and annoying, yes, but it is often right. I can imagine that within Ruff, only the most "right" ones are added anyway, so it's probably a good idea to have the Ruff-filtered Pylint complaints for sure.

The many "too many arguments" complaints that this PR removes, for instance: I think they make a lot of sense to think about. It's very easy to call functions with parameters in the wrong order. Mypy will help a bit with this, but if parameters have the same types, it doesn't. A potential fix is to use keyword-only arguments, but that makes your calls very long. An alternative is to use configuration objects, probably dicts or dataclasses. Dataclasses especially can be nice to pass along configuration between many functions in a consistent way, can check consistency internally with class methods, etc.

Reply:
The pylint implementation of ruff is incomplete, which is why I have not enabled it. I'm afraid it will become too much of a moving target.

I agree that we should think about 'too many arguments', but if the solution is to slap on a `# pylint: too-many-arguments`, then this whole exercise is now a giant anti-pattern that we should try to avoid. It's also not necessarily bad to have too many arguments. The code still runs. Sometimes the work-arounds (i.e. passing around dataclasses and kwargs) are more harmful than what it tries to prevent.

Reply:
Yes true. I also don't like so much that you have to annotate the code to stop the complaints. Ideally, you'd have some central file where you list exceptions to strict checks. Then, perhaps, periodically, or when you are doing a major refactoring operation anyway, you can review all the warnings to easily find design-smells that perhaps you can fix in the same refactoring in one go. I still think it's good to check them, but maybe it could be a separate CI workflow as well. Probably, that one will be red most of the time then, but you can look at it every now and then.